src/digitalIdentity-frontend/src/app/pages/proofTemplate/create-proofTemplate-page/create-proofTemplate-page.component.ts

```diff
@@ -0,0 +1,475 @@
+        this.proofTemplateTmp.attributes[i] = this.proofTemplateTmp.attributes[i + 1];
+        this.proofTemplateTmp.attributes[i].attribID -= 1;
+      }
+      this.proofTemplateTmp.attributes.pop();
+    }
+    (<FormArray>this.proofTemplateFormGroup.controls['attributes']).removeAt(idx);
+  }
+
+  get attributes(): FormArray {
+    return <FormArray>this.proofTemplateFormGroup.get('attributes');
+  }
+
+  createProofTemplateButtonEvent() {
+    //set TmpProofTemplate again
+    this.proofTemplateTmp.name = this.proofTemplateFormGroup.value['name'];
+    this.proofTemplateTmp.version = this.proofTemplateFormGroup.value['version'];
+    for (let elem of this.proofTemplateTmp.attributes) {
+      elem.name =
+        this.proofTemplateFormGroup.value['attributes'][elem.attribID]['name'];
+    }
+
+    //set real proofTemplate
+    this.proofTemplate.name = this.proofTemplateTmp.name;
+    this.proofTemplate.version = this.proofTemplateTmp.version;
+
+    for (let i = 0; i < this.proofTemplateTmp.attributes.length; i++) {
+      if (i >= this.proofTemplate.attributes.length) {
+        this.proofTemplate.attributes.push({
+          // attribID: i,
+          attributeName: ''
+          // value: '',
+          // type: 'String',
+        });
+      }
+      this.proofTemplate.attributes[i].attributeName = this.proofTemplateTmp.attributes[i].name;
+      // this.proofTemplate.attributes[i].type = this.proofTemplateTmp.attributes[i].type;
+      // this.proofTemplate.attributes[i].attribID =
+      //   this.proofTemplateTmp.attributes[i].attribID;
+      // this.proofTemplate.attributes[i].value = this.proofTemplateTmp.attributes[i].value;
+    }
+    let maxi: any =
+      this.proofTemplate.attributes.length - this.proofTemplateTmp.attributes.length;
+    for (let i = 0; i < maxi; i++) {
+      this.proofTemplate.attributes.pop();
+    }
+    let params = this.proofTemplateToHttpParams(this.proofTemplate);
+    // let imageFormData = this.imageToHttpBody(this.proofTemplate); // will be used for sending image file
+    this.postProofTemplate(params);
+  }
+
+  postProofTemplate(params: HttpParams, ): void {
+    this.requestInProgress = true;
+    this.httpService.postRequest(
+      'create proof template',
+      '/proof-template/create',
+      this.proofTemplateFormGroup.value,
+      params
+    )
+      .then((response) => {
+        if (!response.ok) {
+          this.dialogRef.open(InformationPopUpComponent, {
+            data: {
+              header: 'Process failed',
+              text: 'Error ' + response.status + ' \n' + response.error,
+            },
+          });
+          this.requestInProgress = false;
+        } else {
+          this.dialogRef.open(InformationPopUpComponent, {
+            data: {
+              header: 'Creating of proof template was successful',
+              text: 'Server response: ' + response.body,
+            },
+          });
```

Comment:
You should start here with the case of success and then comes the case of failure

Reply:
ok